erpnext/controllers/accounts_controller.py

```diff
@@ -910,10 +914,23 @@ def get_value_in_transaction_currency(self, account_currency, args, field):
 			return flt(args.get(field, 0) / self.get("conversion_rate", 1))
 
 	def validate_qty_is_not_zero(self):
-		if self.doctype != "Purchase Receipt":
-			for item in self.items:
-				if not item.qty:
-					frappe.throw(_("Item quantity can not be zero"))
+		if self.doctype == "Purchase Receipt":
+			return
+
+		title = _("Invalid Quantity")
+		for item in self.items:
+			if not flt(item.qty):
+				frappe.throw(
+					msg=_("Row #{0}: Item quantity cannot be zero").format(item.idx),
+					title=title,
+					exc=InvalidQtyError,
+				)
+			if flt(item.qty) < 0.0 and self.doctype in ["Purchase Order", "Sales Order"]:
+				frappe.throw(
+					msg=_("Row #{0}: Item quantity cannot be negative").format(item.idx),
+					title=title,
+					exc=InvalidQtyError,
+				)
```

Comment:
@marination can we use this property?

<img width="609" alt="Screenshot 2023-11-23 at 10 45 03 PM" src="https://github.com/frappe/erpnext/assets/8780500/e448611b-f822-4855-baf9-5a46c07584df">